### PR TITLE
Lua-Server: Fix hidden bugs

### PR DESCRIPTION
**Pull request by ZenBre4ker**
The lua Server mentioned many globals that were unused, but it seems that they are just misspelled variables in most cases.
THis is untested and is just a theoretical fix. So gotta look what could be wrong now.

### CODE REVIEW

File: gamemodes/terrortown/gamemode/client/cl_voice.lua [hunk 1]
```diff
@@ -178,7 +178,7 @@ function GM:PlayerStartVoice(ply)
 	if not IsValid(g_VoicePanelList) or not IsValid(client) then return end
 
 	-- There'd be an extra one if voice_loopback is on, so remove it.
-	GAMEMODE:PlayerEndVoice(ply, true)
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/client/cl_voice.lua:181]: This change breaks the gamemode

File: gamemodes/terrortown/gamemode/client/cl_voice.lua [hunk 2]
```diff
@@ -297,7 +297,7 @@ local function VoiceClean()
 	for ply, pnl in pairs(PlayerVoicePanels) do
 		if IsValid(pnl) and IsValid(ply) then continue end
 
-		GAMEMODE:PlayerEndVoice(ply)
+		GM:PlayerEndVoice(ply)
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/client/cl_voice.lua:300]: This one as well

File: gamemodes/terrortown/gamemode/server/sv_addonchecker.lua
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/server/sv_addonchecker.lua:1]: Why did you remove addons here?

File: gamemodes/terrortown/gamemode/server/sv_main.lua [hunk 3]
```diff
@@ -1210,7 +1210,7 @@ function BeginRound()
 	-- @realm server
 	hook.Run("TTTBeginRound")
 
-	ents.TTT.TriggerRoundStateOutputs(ROUND_BEGIN)
+	ents.TTT.TriggerRoundStateOutputs(ROUND_ACTIVE)
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/server/sv_main.lua:1213]: Why?

File: gamemodes/terrortown/gamemode/shared/sh_door.lua [hunk 4]
```diff
@@ -232,7 +232,7 @@ if SERVER then
 	function entmeta:SetDoorCanTouchOpen(state, surpressPair)
 		door.SetPlayerCanTouch(self, state)
 
-		self:SetNWBool("ttt2_door_player_touch", PlayerCanTouchDoor(self))
+		self:SetNWBool("ttt2_door_player_touch", door.PlayerCanTouch(self))
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/shared/sh_door.lua:235]: I have to check the code, but I'm pretty sure that change is wrong

**Review by TimGoll — CHANGES_REQUESTED** [review]

File: gamemodes/terrortown/gamemode/server/sv_main.lua (hunk 3 above)
Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/server/sv_main.lua:1213]: There is no ROUND_BEGIN defined

File: gamemodes/terrortown/gamemode/client/cl_voice.lua (hunk 1 above)
Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/client/cl_voice.lua:181]: Why? There is no GAMEMODE:PlayerENdVoice, I checked.

File: gamemodes/terrortown/gamemode/server/sv_addonchecker.lua
Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/server/sv_addonchecker.lua:1]: They were duplicates

File: gamemodes/terrortown/gamemode/shared/sh_door.lua (hunk 4 above)
Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/shared/sh_door.lua:235]: couldnt find anything better, but PLayerCanTouchDoor is not a function anywhere

File: gamemodes/terrortown/gamemode/server/sv_addonchecker.lua
Comment (Histalek) [reply gamemodes/terrortown/gamemode/server/sv_addonchecker.lua:1]: Maybe we should order these by ID?

File: gamemodes/terrortown/gamemode/server/sv_main.lua (hunk 3 above)
Comment (Histalek) [reply gamemodes/terrortown/gamemode/server/sv_main.lua:1213]: `ROUND_BEGIN` was **never** defined (at least not in sh_init.lua)

File: gamemodes/terrortown/gamemode/client/cl_voice.lua (hunk 1 above)
Comment (EntranceJew) [reply gamemodes/terrortown/gamemode/client/cl_voice.lua:181]: image

Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/client/cl_voice.lua:181]: Thats GM:PlayerEndVoice for which I exchanged this.

File: gamemodes/terrortown/gamemode/client/cl_voice.lua [hunk 5]
```diff
@@ -178,7 +178,7 @@ function GM:PlayerStartVoice(ply)
 	if not IsValid(g_VoicePanelList) or not IsValid(client) then return end
 
 	-- There'd be an extra one if voice_loopback is on, so remove it.
-	GAMEMODE:PlayerEndVoice(ply, true)
+	GM:PlayerEndVoice(ply, true)
```
Comment (ZenBre4ker) [inline gamemodes/terrortown/gamemode/client/cl_voice.lua:181]: See one line below please.

File: gamemodes/terrortown/gamemode/client/cl_voice.lua (hunk 1 above)
Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/client/cl_voice.lua]: WHile I dont get it, GAMEMODE is the actual working hook...... I reseted it

Comment (EntranceJew) [reply gamemodes/terrortown/gamemode/client/cl_voice.lua]: GAMEMODE is defined at the first pass when the gamemode is loaded, GM is only available after that

Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/client/cl_voice.lua]: > GAMEMODE is defined at the first pass when the gamemode is loaded, GM is only available after that

Is this anywhere documented?

Comment (TimGoll) [reply gamemodes/terrortown/gamemode/client/cl_voice.lua]: I'm pretty sure it is tge other was around. GM is defined on first pass, GAMEMODE, when the game is running

File: gamemodes/terrortown/gamemode/shared/sh_door.lua (hunk 4 above)
Comment (TimGoll) [reply gamemodes/terrortown/gamemode/shared/sh_door.lua:235]: I used git blame a bit and I guess you're right. At some point we moved these local functions into global functions.

old:
image

new:
image

Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/shared/sh_door.lua:235]: Good to know, I tested a lot and those functions are never called. They are not even referenced inside TTT2, might they have become useless?

File: gamemodes/terrortown/gamemode/server/sv_main.lua (hunk 3 above)
Comment (TimGoll) [reply gamemodes/terrortown/gamemode/server/sv_main.lua:1213]: You're right. These are the options:

```
ROUND_WAIT = 1
ROUND_PREP = 2
ROUND_ACTIVE = 3
ROUND_POST = 4
```

`ROUND_ACTIVE` should be the correct choice. This bug existed (and still exists(!)) in base TTT: 
https://github.com/Facepunch/garrysmod/blob/89b23437b5075e7efd5c379e74ccbf534679bcea/garrysmod/gamemodes/terrortown/gamemode/init.lua#L695

This means that the entity output is never triggered:
https://github.com/Facepunch/garrysmod/blob/master/garrysmod/gamemodes/terrortown/entities/entities/ttt_map_settings.lua#L59-L68

This fix might trigger some things on maps that have *never* been triggered. We should test this, even though it is a bugfix

File: gamemodes/terrortown/gamemode/shared/sh_door.lua (hunk 4 above)
Comment (TimGoll) [reply gamemodes/terrortown/gamemode/shared/sh_door.lua:235]: such as `door.PlayerCanTouch`? These are used for targetID by setting the NW bool on the server. They are also an external API that can be used by addons - if they want to. So I don't see why we shouldn't keep them.

Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/shared/sh_door.lua:235]: grafik
These 3 here are not called anywhere, might they be deprecated?

Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/shared/sh_door.lua:235]: I meant entmeta:SetDoorCanTouchOpen and CanUseOpen and AutoCloses

Comment (TimGoll) [reply gamemodes/terrortown/gamemode/shared/sh_door.lua:235]: I don't think I intended on using them in TTT2, I added them as part of the API so addons can use this on entities. I don't know if any addon uses this, but I wouldn't say that it is deprecated.

I know for a fact that i use `entmeta:ToggleDoor` in one of my addons for example. As well as `entmeta:LockDoor`. I therefore exposed all settings to the API, instead of just those I needed at the time.

Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/shared/sh_door.lua:235]: > I don't think I intended on using them in TTT2, I added them as part of the API so addons can use this on entities. I don't know if any addon uses this, but I wouldn't say that it is deprecated.
> 
> I know for a fact that i use `entmeta:ToggleDoor` in one of my addons for example. As well as `entmeta:LockDoor`. I therefore exposed all settings to the API, instead of just those I needed at the time.

Yeah those I tested. Okay well, thanks for the input ❤️

Comment (TimGoll) [reply gamemodes/terrortown/gamemode/shared/sh_door.lua:235]: > I meant entmeta:SetDoorCanTouchOpen and CanUseOpen and AutoCloses

It is probably not uses right now, yes. But because this is a general API for doors, I don't see how it is deprecated.

One could make an addon that is an admin tools to setup doors for example

**Comment by TimGoll** [comment]
> I tested these changes on a single per-change basis. At least I didnt break anything. Should be good to go.

I'm mostly fine with your changes. But why did you remove items from the addon checker?

File: gamemodes/terrortown/gamemode/server/sv_addonchecker.lua
Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/server/sv_addonchecker.lua:1]: @TimGoll I removed items that were duplicated by steamID

Comment (TimGoll) [reply gamemodes/terrortown/gamemode/server/sv_addonchecker.lua:1]: I missed the replies hier. Then I'm fine with it

**Review by TimGoll — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,12 @@ All notable changes to TTT2 will be documented here. Inspired by [keep a changel
 
 ## Unreleased
 
+### Fixed
+
+- Fixed DNA Tester now actually adding new samples
+- TraitorButton config files should now actually work
+- Entities on roundstart are now triggered with the actual value of ROUND_ACTIVE
```
Comment (TimGoll) [inline CHANGELOG.md:11]: It's not about general map entities, the entity in question is the `ttt_map_settings` entity

**Comment by TimGoll** [comment]
Still a bit unprecise. The map_settings entity (the actual entity on the map) was never triggered before because the check did not pass

**Comment by ZenBre4ker** [comment]
> Still a bit unprecise. The map_settings entity (the actual entity on the map) was never triggered before because the check did not pass

ACtually I just saw the implementation, this change does nothing at all. If the state is nil it just gets the current round_state

**Comment by TimGoll** [comment]
> > Still a bit unprecise. The map_settings entity (the actual entity on the map) was never triggered before because the check did not pass
> 
> ACtually I just saw the implementation, this change does nothing at all. If the state is nil it just gets the current round_state

Ahh, well, then it didn't really change anything I guess

**Review by saibotk — APPROVED** [review]
Looks good but you did not mention the other fixes in the changelog

**Comment by ZenBre4ker** [comment]
yeah the rest are not really worth mentioning, some minor changes.
